addons/binding/org.openhab.binding.folderwatcher/src/main/java/org/openhab/binding/folderwatcher/internal/handler/FolderLocalWatcherHandler.java

```diff
@@ -0,0 +1,169 @@
+
+    public FolderLocalWatcherHandler(Thing thing) {
+        super(thing);
+
+        currentLocalListingFile = new File(ConfigConstants.getUserDataFolder() + File.separator + "FolderWatcher"
+                + File.separator + thing.getUID().getAsString().replace(':', '_') + ".data");
+
+        try {
+            previousLocalListing = WatcherCommon.initStorage(currentLocalListingFile);
+        } catch (IOException e) {
+            logger.debug("Can't write file: ", currentLocalListingFile, e);
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("handleCommand for channel {}: {}", channelUID.getId(), command.toString());
+        logger.debug("Command sending not supported by this binding");
+    }
+
+    @Override
+    public void initialize() {
+
+        config = getConfigAs(FolderLocalWatcherConfiguration.class);
+        updateStatus(ThingStatus.UNKNOWN);
+
+        if (config.pollIntervalLocal > 0) {
+            updateStatus(ThingStatus.ONLINE);
+            executionJob = scheduler.scheduleWithFixedDelay(periodicExecutionRunnable, config.pollIntervalLocal,
+                    config.pollIntervalLocal, TimeUnit.SECONDS);
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Polling interval can't be null or negative");
+        }
+
+    }
+
+    @Override
+    public void dispose() {
+        executionJob.cancel(true);
+
+    }
+
+    ArrayList<String> listFiles(String path, boolean listHidden, boolean recursive) throws IOException {
+
+        File[] fileList;
+        ArrayList<String> dirList = new ArrayList<String>();
+
+        // create new file
+        final File f = new File(path);
+
+        // array of files and directory
+        fileList = f.listFiles(filter -> listHidden || !filter.isHidden());
+
+        // for each name in the path array
+        if (fileList != null && fileList.length > 0) {
+            for (File file : fileList) {
+
+                if (file.isDirectory()) {
+
+                    if (recursive) {
+                        dirList.addAll(listFiles(file.getAbsolutePath(), listHidden, recursive));
+
+                    }
+                }
+                if (file.isFile()) {
+                    dirList.add(file.getAbsolutePath());
+                }
+            }
+        }
+
+        return dirList;
+    }
+
+    protected Runnable periodicExecutionRunnable = new Runnable() {
+        @Override
+        public void run() {
+
+            String RootDir = config.localDir;
+            ArrayList<String> diffLocalListing = new ArrayList<String>();
+
+            currentLocalListing.clear();
+
+            try {
+                currentLocalListing.addAll(listFiles(RootDir, config.listHiddenLocal, config.listRecursiveLocal));
+            } catch (IOException e) {
+
+                logger.debug("Can't read directory: ", config.localDir, e);
```

Comment:
This logging is missing *one* format parameter

Reply:
Done